tests/services/test_file_workflow_service.py

```
@@ -559,3 +572,378 @@ def test_recursive_scandir(tmp_path):
+    os.utime(new_file, (now - 60, now - 60))
+
+    # Run the workflow
+    service = FileWorkflowService()
+    
+    # We need to mock SessionFactory because process_single_file opens its own sessions.
+    # By returning the same db_session (wrapped in a lambda), we ensure it sees 
+    # the tables created by the db_session fixture.
+    # We also mock SessionLocal used by some other services if any.
+    with patch("app.services.file_workflow_service.SessionFactory", side_effect=lambda: db_session):
+        # We also need to mock scan_progress_manager to avoid singleton state issues
+        with patch("app.services.file_workflow_service.scan_progress_manager") as mock_progress:
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)